example/otel-collector/main.go

```diff
@@ -47,11 +47,11 @@ func initProvider() func() {
 	// `localhost:30080` address. Otherwise, replace `localhost` with the
 	// address of your cluster. If you run the app inside k8s, then you can
 	// probably connect directly to the service through dns
-	exp, err := otlp.NewExporter(
-		otlp.WithInsecure(),
+	config := otlp.NewConnectionConfig(otlp.WithInsecure(),
 		otlp.WithAddress("localhost:30080"),
 		otlp.WithGRPCDialOption(grpc.WithBlock()), // useful for testing
 	)
+	exp, err := otlp.NewExporter(config, config)
```

Comment:
NewExporter takes in two arguments: a configuration for the metrics connection and one for the traces

Reply:
Can this use functional options instead? That would allow a user to choose to configure only metrics or traces and could also provide a convenience option for using the same connection configuration for both.  E.g.:

```go
// only provide a ConnectionConfiguration for metrics, traces would not be exported
exp, err := otlp.NewExporter(otlp.WithMetricsConnection(config))
```

or

```go
// use the same ConnectionConfiguration for both signal types
exp, err := otlp.NewExporter(otlp.WithConnection(config))
```

Reply:
True! But the NewExporter needs to take in a structure of the form:
```golang
type ConnConfigurations struct {
	metrics config
	traces  config
}
```

And then you'd have more functional methods like:
```golang
func NewConnConfigurations() ConnConfigurations
func (occ ConnConfigurations) WithConnection(opts ...ExporterOption) ConnConfigurations
func (occ ConnConfigurations) WithMetricsConnection(opts ...ExporterOption) ConnConfigurations
func (occ ConnConfigurations) WithTracesConnection(opts ...ExporterOption) ConnConfigurations
```
Resulting in client calls like so:
```golang
exp, err := otlp.NewExporter(otlp.NewConnConfigurations().
		WithTracesConnection(otlp.WithTLSCredentials(creds)))
```
Is this what you are thinking?

Reply:
@Aneurysm9 a problem with the way you suggested is that if someone wants to configure different trace and metrics options, you'd still have a method like `WithMetricsAndTracesConfigurations(config, config)`. Or is there something I am missing?

I update the interface a bit, to make it more declarative. I think that this works quite well.
It uses a structure holding configuration for both metrics and traces:

```go
type ConnConfigurations struct {
	metrics config
	traces  config
}
```

Then adds a declarative interface on top, allowing users to configure common settings and custom settings on the specific signals:
```go
// NewConnections initializes a config struct with default values and applies
// any ExporterOptions provided for both metrics and traces.
func NewConnections(opts ...ExporterOption) ConnConfigurations

// SetMetricOptions updates the connection configuration for the metrics endpoint,
// overriding options set with NewConnections, or setting new metrics specific options.
func (occ ConnConfigurations) SetMetricOptions(opts ...ExporterOption) ConnConfigurations

// SetTraceOptions updates the connection configuration for the traces endpoint,
// overriding options set with NewConnections, or setting new trace specific options.
func (occ ConnConfigurations) SetTraceOptions(opts ...ExporterOption) ConnConfigurations
```
There's some test cases for this in `exporters/otlp/options_test.go` showing the uses.

Looking forward to hearing your feedback!